railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -141,6 +141,7 @@ def test
     end
 
     def tmp
+      empty_directory_with_keep_file "tmp"
```

Comment:
I wouldn't mind about older apps because the error is pretty obvious, but as you guys wish.

At least I'd make the restart task depend on another task that makes the mkdir_p and would call that task something like add_tmp_for_broken_cloned_apps or something like that